clusterctl/examples/google/generate-yaml.sh

```diff
@@ -4,9 +4,40 @@ set -e
 GCLOUD_PROJECT=$(gcloud config get-value project)
 ZONE=$(gcloud config get-value compute/zone)
 ZONE="${ZONE:-us-central1-f}"
+CLUSTER_NAME=test1
+
+MACHINE_TEMPLATE_FILE=machines.yaml.template
+MACHINE_GENERATED_FILE=machines.yaml
+CLUSTER_TEMPLATE_FILE=cluster.yaml.template
+CLUSTER_GENERATED_FILE=cluster.yaml
+PROVIDERCOMPONENT_TEMPLATE_FILE=provider-components.yaml.template
+PROVIDERCOMPONENT_GENERATED_FILE=provider-components.yaml
+ADDON_TEMPLATE_FILE=addons.yaml.template
+ADDON_GENERATED_FILE=addons.yaml
+
+MACHINE_CONTROLLER_SA_FILE=machine-controller-serviceaccount.json
+MACHINE_CONTROLLER_SA_NAME="machine-controller-$CLUSTER_NAME"
+MACHINE_CONTROLLER_SA_EMAIL="$MACHINE_CONTROLLER_SA_NAME@$GCLOUD_PROJECT.iam.gserviceaccount.com"
+MACHINE_CONTROLLER_SA_KEY=
+LOADBALANCER_SA_FILE=loadbalancer-serviceaccount.json
+LOADBALANCER_SA_NAME="loadbalancer-$CLUSTER_NAME"
+LOADBALANCER_SA_EMAIL="$LOADBALANCER_SA_NAME@$GCLOUD_PROJECT.iam.gserviceaccount.com"
+LOADBALANCER_SA_KEY=
+
+# TODO: The following service accounts will eventually be provisioned by the cluster controller. In the meanwhile, they are provisioned here.
+MASTER_SA_NAME="master-$CLUSTER_NAME"
+MASTER_SA_EMAIL="$MASTER_SA_NAME@$GCLOUD_PROJECT.iam.gserviceaccount.com"
+WORKER_SA_NAME="worker-$CLUSTER_NAME"
+WORKER_SA_EMAIL="$WORKER_SA_NAME@$GCLOUD_PROJECT.iam.gserviceaccount.com"
+
+MACHINE_CONTROLLER_SSH_PUBLIC_FILE=machine-controller-key.pub
+MACHINE_CONTROLLER_SSH_PUBLIC=
+MACHINE_CONTROLLER_SSH_PRIVATE_FILE=machine-controller-key
+MACHINE_CONTROLLER_SSH_PRIVATE=
+MACHINE_CONTROLLER_SSH_USER_PLAIN=clusterapi
+MACHINE_CONTROLLER_SSH_USER=$(echo -n "$MACHINE_CONTROLLER_SSH_USER_PLAIN" | base64 -w0)
+
 
-TEMPLATE_FILE=machines.yaml.template
-GENERATED_FILE=machines.yaml
 OVERWRITE=0
```

Comment:
OVERWRITE is always set to 0, is it meant for testing or is the user allowed to specify this to force an overwrite?

Reply:
User can specify a flag to force overwrite later in the script.

Reply:
It does not overwrite by default as a safety mechanism.